### PR TITLE
Add body_regex option for deleting comments

### LINKED ISSUE #37: Node.js 12 actions are deprecated
Node.js 12 actions are deprecated. Please update the following actions to use Node.js 16: izhangzhihao/delete-comment@master. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/.

### PR DESCRIPTION
**Pull request by brtknr**
This PR proposes the following changes:
- Add a body_regex option for deleting comments.
- Make issue_number mandatory so that we don't automatically traverse all issues by mistake.
- Upgrade to use node 16 since node 12 is deprecated. Fixes #37

### CODE REVIEW

**Comment by brtknr** [comment]
Not sure why the build is still failing

File: package.json [hunk 1]
```diff
@@ -31,9 +31,9 @@
   },
   "devDependencies": {
     "@types/jest": "^26.0.23",
-    "@types/node": "^15.0.1",
+    "@types/node": "^15.0.1",o
```
Comment (izhangzhihao) [inline package.json:34]:
```suggestion
    "@types/node": "^15.0.1",
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -11,7 +11,13 @@ jobs:
...
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+      - name: Setup NodeJS
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+      - run: |
+          export NODE_OPTIONS=--openssl-legacy-provider
```
Comment (izhangzhihao) [inline .github/workflows/test.yml:20]:
```suggestion
      - run: |
          export NODE_OPTIONS=--openssl-legacy-provider
```

**Comment by izhangzhihao** [comment]
issue_number mandatory so that we don't automatically traverse all issues by mistake is a good option. Sometimes I want to delete all comments for all issues, I think that could be an option as well.

**Comment by brtknr** [comment]
Screenshot 2023-06-29 at 20 48 52
Also I noticed this behaviour, I think the GITHUB_TOKEN is scoped to the issue that created the pull request so it traverses the same pull request repeatedly rather than other issues in the repo.

**Comment by brtknr** [comment]
Screenshot 2023-06-29 at 22 01 05

Also I noticed the above behaviour, in a repo with large number of closed issues, it traverses all closed issues and eventually hits API rate limit error which is not a great user experience.

**Review by izhangzhihao — APPROVED** [review]
LGTM

**Comment by brtknr** [comment]
Would be great to tag this as a v1 release

**Pull request merged**